modules/rpc/yarpc_test.go

```diff
@@ -36,30 +35,36 @@ func TestRegisterDispatcher_OK(t *testing.T) {
 	RegisterDispatcher(defaultYARPCDispatcher)
 }
 
+func TestRegisterStarter_OK(t *testing.T) {
+	t.Parallel()
+	RegisterStarter(defaultYARPCStarter)
+}
+
 func TestDispatcher(t *testing.T) {
 	t.Parallel()
 	c := dispatcherController{}
 	host := service.NopHost()
-	require.NoError(t, c.addConfig(yarpcConfig{AdvertiseName: host.Name(), inbounds: []transport.Inbound{}}))
+	c.addConfig(yarpcConfig{transports: transports{inbounds: []transport.Inbound{}}})
 	assert.NoError(t, c.Start(host))
 }
 
 func TestDifferentAdvertiseNameReturnsError(t *testing.T) {
 	t.Parallel()
 	c := dispatcherController{}
 	cfg := yarpcConfig{
-		AdvertiseName: "fx",
-		inbounds:      []transport.Inbound{http.NewTransport().NewInbound("")},
+		transports: transports{
+			inbounds: []transport.Inbound{http.NewTransport().NewInbound("")},
+		},
 	}
 
-	require.NoError(t, c.addConfig(cfg))
+	c.addConfig(cfg)
 	assert.Error(t, c.Start(service.NopHost()))
```

Comment:
True way is to compare error messages, but here we expect a system error, can't open a default port.

Changed port to be -1, when 80 is open, all I want is to trigger transports error.